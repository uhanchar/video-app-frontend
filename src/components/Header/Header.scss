@import "src/constants";

.header {
  width: 100%;
  height: $header-height;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-sizing: border-box;
  padding: 0 1.25rem;
  background: #ffffff;

  > .title {
    color: $primary-color-black;
    font-size: 1.5rem;
    text-decoration: none;
  }
}
