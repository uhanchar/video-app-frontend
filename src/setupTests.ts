// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';

export const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  // @ts-ignore
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));
