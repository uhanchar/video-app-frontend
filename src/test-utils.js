import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const reducerInitialState = {};

function render(
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';

export { render };
