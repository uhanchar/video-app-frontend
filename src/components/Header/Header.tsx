import React, { FC } from 'react';
import { NavLink, useHistory } from 'react-router-dom';
import { Button } from '@material-ui/core';

import 'components/Header/Header.scss';
import { LocationPath } from 'constants/location-path.enum';

const Header: FC = () => {
  const history = useHistory();

  const onNavigateToUploadVideoPageButtonClick = () => {
    history.push(LocationPath.Upload);
  };

  return (
    <header className="header" data-testid="header">
      <NavLink to={LocationPath.Home} className="title">Video app</NavLink>

      <Button
        variant="contained"
        color="primary"
        size="small"
        data-testid="header-navigate-button"
        onClick={onNavigateToUploadVideoPageButtonClick}
      >
        Upload video
      </Button>
    </header>
  );
};

export default Header;
