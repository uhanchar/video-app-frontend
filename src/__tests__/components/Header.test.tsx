import React from 'react';
import { fireEvent } from '@testing-library/react';

import { render as renderWithRedux } from 'test-utils';
import Header from 'components/Header/Header';
import { LocationPath } from 'constants/location-path.enum';
import { mockHistoryPush } from 'setupTests';

const title = 'Video app';

describe('test for Header component', () => {
  it('should render Header component', () => {
    const { asFragment } = renderWithRedux(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Loader title content', () => {
    const { getByTestId } = renderWithRedux(<Header />);

    expect(getByTestId('header')).toHaveTextContent(title);
  });

  it('should change url on upload video button click', () => {
    const { getByTestId } = renderWithRedux(<Header />);

    fireEvent.click(getByTestId('header-navigate-button'));

    expect(mockHistoryPush).toHaveBeenCalledWith(LocationPath.Upload);
  });
});
