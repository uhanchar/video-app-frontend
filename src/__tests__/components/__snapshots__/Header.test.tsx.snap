// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Header component should render Header component 1`] = `
<DocumentFragment>
  <header
    class="header"
    data-testid="header"
  >
    <a
      aria-current="page"
      class="title active"
      href="/"
    >
      Video app
    </a>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
      data-testid="header-navigate-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Upload video
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </header>
</DocumentFragment>
`;
